src/sdk/object_sdk.js

```diff
@@ -233,9 +233,10 @@ class ObjectSDK {
             ns_info.endpoint_type === 'S3_COMPATIBLE' ||
             ns_info.endpoint_type === 'FLASHBLADE' ||
             ns_info.endpoint_type === 'IBM_COS') {
-            const httpOptions = (ns_info.endpoint_type === 'AWS') ? undefined : {
-                agent: http_utils.get_unsecured_http_agent(ns_info.endpoint)
-            };
+
+            const agent = http_utils.get_http_agent(ns_info.endpoint, {
+                rejectUnauthorized: ns_info.endpoint_type !== 'AWS'
```

Comment:
This is incorrect. If we are talking to AWS we want `rejectUnauthorized = true` in this case we will have false.